includes/class-wc-cart.php

```diff
@@ -1600,9 +1600,9 @@ public function check_customer_coupons( $posted ) {
 					if ( $coupon->is_valid() ) {
 
 						// Limit to defined email addresses
-						if ( is_array( $coupon->customer_email ) && sizeof( $coupon->customer_email ) > 0 ) {
+						if ( is_array( $coupon->get_email_restrictions() ) && sizeof( $coupon->get_email_restrictions() ) > 0 ) {
 							$check_emails           = array();
-							$coupon->customer_email = array_map( 'sanitize_email', $coupon->customer_email );
+							$coupon->set_email_restrictions( array_map( 'sanitize_email', $coupon->get_email_restrictions() ) );
```

Comment:
Perhaps consider the sanitize within the setter, since anyone can send data there and we should keep it valid?